apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/Trace/DelegatingTracer.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright 2017 Google Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Api.Gax;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading.Tasks;
+
+namespace Google.Cloud.Diagnostics.Common
+{
+    /// <summary>
+    /// An <see cref="IManagedTracer"/> that delegeates all calls to another <see cref="IManagedTracer"/>
+    /// that is retrieved from the given function.
+    /// </summary>
+    internal class DelegatingTracer : IManagedTracer
+    {
+        private readonly Func<IManagedTracer> _managedTracerFactory;
+
+        internal DelegatingTracer(Func<IManagedTracer> managedTracerFactory)
```

Comment:
`...Factory` implies creating new instances on each invocation. I would probably call this `managedTracerGetter` or something like that instead.